File: front-end/src/app/product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Product } from './product';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  private baseUrl = 'http://localhost:8080/api';

  constructor(private http: HttpClient) { }

  getProduct(id: number): Observable<any> {
    return this.http.get(this.baseUrl+'/products/'+id);
  }

  createProduct(product: Object): Observable<Object> {
    return this.http.post(this.baseUrl+'/products/', product);
  }

  updateProduct(product: Object): Observable<Object> {
    return this.http.put(this.baseUrl+'/products/', product);
  }

  deleteProduct(id: number): Observable<any> {
    return this.http.delete(this.baseUrl+'/products/'+id);
  }

  getProductsList(): Observable<any> {
    return this.http.get(this.baseUrl+'/products/');
  }

}

File: back-end/src/main/java/com/company/service/ProductService.java
package com.company.service;

import com.company.model.Product;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ProductService {
	Flux<Product> getProducts();
	Mono<Product> getProduct(Long id);
	Mono<Product> createProduct(Product product);
	Mono<Product> updateProduct(Product product);
	Mono<Void> deleteProduct(Long id);
}
